nxcomp/src/Loop.cpp

```diff
@@ -4278,15 +4278,18 @@ int ListenConnectionTCP(const char *host, long port, const char *label)
 
 int ListenConnection(ChannelEndPoint &endpoint, const char *label)
 {
-  char *unixPath, *host;
+  char *unixPath = NULL, *host = NULL;
   long port;
+  int result = -1;
   if (endpoint.getUnixPath(&unixPath)) {
-    return ListenConnectionUnix(unixPath, label);
+    result = ListenConnectionUnix(unixPath, label);
   }
   else if (endpoint.getTCPHostAndPort(&host, &port)) {
-    return ListenConnectionTCP(host, port, label);
+    result = ListenConnectionTCP(host, port, label);
   }
-  return -1;
+  free(unixPath);
```

Comment:
Set them to `NULL` for good measure.

Reply:
they are local and it is at the end of the function, I really don't see a point in NULLing here.

Reply:
It's always a good (defensive) measure to set `free`d data to NULL. It's a very cheap operation and prevents errors like exactly these if code is ever added after the `free` call.

A lot of software defines a macro such as
```
#define SAFE_FREE(ptr) do { free(ptr); ptr = NULL; } while (0)
```

and uses them to free memory.

Reply:
Then let's introduce this macro, too. But why the while "loop"?

Reply:
https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros